src/index.ts

```diff
@@ -55,9 +63,15 @@ export class PromiseQueue {
 		}
 	}
 	public add<T>(fn: () => T | PromiseLike<T>): Promise<T> {
+		this.metrics.emit('arrival');
+		this.metrics.emit('queueLength', this.queue.length);
+		this.metrics.emit('inFlight', this.inFlight);
+		const enqueueTime = process.hrtime();
```

Comment:
Can you move this inside the `new Promise` call to limit it to just the scope it's needed

Reply:
done